api/src/wfl/source.clj

```diff
@@ -285,13 +287,34 @@
            (run! #(write-snapshot-id source %)))
       (log/debug (format "%s Running snapshot jobs updated." (log-prefix source))))))
 
+;; Workloads in general may be updated more frequently,
+;; but overpolling the TDR increases the chances of:
+;; - creating many single-row / low-cardinality snapshots
+;; - locking the dataset / running into dataset locks
+(def ^:private tdr-source-polling-interval-minutes 20)
+
+;; TODO: add unit tests
+(defn ^:private tdr-source-should-poll?
+  "Return true if it's been at least `tdr-source-polling-interval-minutes`
+   since `last_checked` -- when we last checked for new rows in the TDR."
+  [{:keys [type id last_checked] :as _source} utc-now]
+  (let [checked            (timestamp-to-offsetdatetime last_checked)
+        minutes-since-poll (.between ChronoUnit/MINUTES checked utc-now)]
+    (log/debug {:type               type
```

Comment:
Wouldn't mind talking more about this.  It will get emitted every 20 seconds, not sure if that's too frequent, even for a DEBUG.

At any rate, was nice to experiment with passing something other than a formatted string to the new logger (also something I wouldn't mind talking more about).